flexmeasures/data/services/sensors.py

```diff
@@ -1,43 +1,34 @@
 from __future__ import annotations
 
-from werkzeug.exceptions import NotFound
+import sqlalchemy as sa
 
 from flexmeasures import Sensor, Account
 from flexmeasures.data.models.generic_assets import GenericAsset
 
 
-def get_sensors(
-    account_name: str | None = None,
+def get_account_sensors(
+    accounts: list[Account | None],
+    sensor_ids: list[int] | None = None,
 ) -> list[Sensor]:
-    """Return a list of Sensor objects.
-
-    :param account_name: optionally, filter by account name.
-    """
-    sensor_query = Sensor.query
-
-    if account_name is not None:
-        account = Account.query.filter(Account.name == account_name).one_or_none()
-        if not account:
-            raise NotFound(f"There is no account named {account_name}!")
-        sensor_query = (
-            sensor_query.join(GenericAsset)
-            .filter(Sensor.generic_asset_id == GenericAsset.id)
-            .filter(GenericAsset.owner == account)
-        )
-
-    return sensor_query.all()
-
-
-def get_public_sensors(sensor_ids: list[int] | None = None) -> list[Sensor]:
-    """Return a list of Sensor objects that belong to a public asset.
+    """Return a list of Sensor objects that belong to any of the given accounts.
 
+    :param accounts: select only sensors from this list of accounts
+                     (include None to select sensors that belong to a public asset).
     :param sensor_ids: optionally, filter by sensor id.
     """
-    sensor_query = (
-        Sensor.query.join(GenericAsset)
-        .filter(Sensor.generic_asset_id == GenericAsset.id)
-        .filter(GenericAsset.account_id.is_(None))
+    account_ids = [account.id for account in accounts if account is not None]
+    sensor_query = Sensor.query.join(GenericAsset).filter(
+        Sensor.generic_asset_id == GenericAsset.id
     )
+    if None in accounts:
```

Comment:
Confusing that one might pass [AccountA, AccountB, None] and only get back public sensors.

Simpler (no support for mixing): do not pass accounts at all to get public sensors. None cannot be part of the list.

Reply:
> Confusing that one might pass [AccountA, AccountB, None] and only get back public sensors.

In this case you should get back public sensors as well as sensors from both accounts. Is the documentation not clear, or is there a problem with the code?

> Simpler (no support for mixing): do not pass accounts at all to get public sensors. None cannot be part of the list.

That wouldn't yield the same benefit, because `sensors_to_show()` would need to call the function twice, resulting in two queries instead of one.

Reply:
I overlooked line 26, sorry.